src/gc/os.d

```diff
@@ -141,3 +141,54 @@ else
 {
     static assert(false, "No supported allocation methods available.");
 }
+
+/**
+   Check for any kind of memory pressure.
+
+   Params:
+      mapped = the amount of memory mapped by the GC in bytes
+   Returns:
+       true if memory is scarce
+*/
+// TOOD: get virtual mem sizes and current usage from OS
+// TODO: compare current RSS and avail. physical memory
+version (Windows)
+{
+    bool isLowOnMem(size_t mapped) nothrow @nogc
+    {
+        version (D_LP64)
+            return false;
+        else
+        {
+            import core.sys.windows.windows;
+            MEMORYSTATUS stat;
+            GlobalMemoryStatus(&stat);
+            // Less than 5 % of virtual address space available
+            return stat.dwAvailVirtual < 0.05 * stat.dwTotalVirtual;
+        }
+    }
+}
+else version (OSX)
+{
+    bool isLowOnMem(size_t mapped) nothrow @nogc
+    {
+        enum size_t GB = 2 ^^ 30;
+        version (D_LP64)
+            return false;
+        else
+            // 80 % of available 4GB is used for GC (excluding malloc and mmap)
+            return mapped > 0.8 * 4 * GB;
```

Comment:
This actually generates a floating point comparison which is a bit odd. Maybe better use `8 * 4 * (GB / 10)` or slightly inaccurate `3 * GB`?